src/test/java/bio/terra/service/dataset/DatasetSchemaUpdateValidatorTest.java

```diff
@@ -150,4 +157,343 @@ void testSchemaUpdateWithDuplicateRelationships() throws Exception {
         errorModel.getErrorDetail().get(0),
         containsString("DuplicateRelationshipNames"));
   }
+
+  @Test
+  void testNoColumns() throws Exception {
+    String newTableName = "new_table";
+    DatasetSchemaUpdateModel updateModel =
+        new DatasetSchemaUpdateModel()
+            .description("No columns with new table")
+            .changes(
+                new DatasetSchemaUpdateModelChanges()
+                    .addTables(List.of(DatasetFixtures.tableModel(newTableName, List.of()))));
+    ErrorModel errorModel = expectBadDatasetUpdateRequest(updateModel);
+    assertThat(
+        "No columns with new table throws error",
+        errorModel.getErrorDetail().get(0),
+        containsString("IncompleteSchemaDefinition"));
+  }
+
+  @Test
+  void testInvalidDatatypeColumns() throws Exception {
+    // Load bad data type from JSON b/c couldn't set it in the model
+    String json = TestUtils.loadJson("update-schema-bad-data-type.json");
+    ErrorModel errorModel = expectBadDatasetUpdateRequest(json);
+    String responseBody = String.join(", ", errorModel.getErrorDetail());
+    assertThat(
+        "Invalid DataTypes are logged and returned",
+        responseBody,
+        Matchers.containsString(
```

Comment:
This is already a static import for this, I believe `Matchers.containsString()` is the same as `CoreMatchers.containsString()`
```suggestion
        containsString(
```